packages/destination-actions/src/destinations/metronome/sendEvent/index.ts

```diff
@@ -0,0 +1,85 @@
+import type { ActionDefinition } from '@segment/actions-core'
+import type { Settings } from '../generated-types'
+import type { Payload } from './generated-types'
+
+function serializeEvent(event: Payload) {
+  return {
+    ...event,
+    // convert the timestamp into rfc3999 format
+    timestamp: new Date(event.timestamp).toISOString(),
+  }
+}
+
+const action: ActionDefinition<Settings, Payload> = {
+  title: 'Send Event',
+  description: 'Send an event to Metronome',
+  fields: {
+    transaction_id: {
+      type: 'string',
+      label: 'transaction_id',
+      description: 'The Metronome transaction ID uniquely identifies an event to ensure Metronome only processes each event once.',
+      required: true,
+      default: {
+        '@path': '$.messageId'
+      }
+    },
+    customer_id: {
+      type: 'string',
+      label: 'customer_id',
+      description: 'The Metronome customer ID or ingest alias this event should be associated with.',
+      required: true,
+      default: {
+        // By default, use the group ID if it exists, otherwise use the customer ID. If neither exist, use the anonymous ID.
+        '@if': {
+          exists: { '@path': '$.groupId' },
```

Comment:
Can you explain what this @path syntax is doing?
Also, how come we are looking for groupId? Where is that coming from?
The comment also suggests we are going to use the customer ID but this is looking for userId. Are those the same thing?

Reply:
Idk what the @ path syntax is besides being how other action destinations work. Basically it allows us set smart defaults for the mapping the user will do between the segment event https://segment.com/docs/connections/spec/common/ and our fields. It's not documented but there _are_ TS types for it and all the other destinations use it, so I assume documentation will come.

`groupId` from Segment is described as 

> Group / Account ID. 
> This is useful in B2B use cases where you need to attribute your non-group calls to a company or account. It is relied on by several Customer Success and CRM tools.

and 

> The group API call is how you associate an individual user with a group—be it a company, organization, account, project, team or whatever other crazy name you came up with for the same concept!

Basically, for the ideal Segment user, the `userId` would be the ID of the person taking the action, and the `groupId` would the the team or org they're in. `anonymousId` is some hopefully persistent ID that could change at any time. 

As such, we use those fields (in order) as the Metronome customer ID, since the order roughly reflects how useful they'd be to us. 

This is just a default for the UI, when setting up Metronome in Segment you'd have to explicitly choose how you want to populate `customer_id` given your data, so it doesn't matter if this isn't exactly right - It's expected that users choose the right setting here, this is just meant to be a helpful default.